lib/chef/formatters/doc.rb

```diff
@@ -258,6 +259,21 @@ def resource_action_start(resource, action, notification_type = nil, notifier =
         indent
       end
 
+      def resource_update_progress(resource, current, total, interval)
+        @progress[resource] ||= 0
+
+        percent_complete = (current.to_f / total.to_f * 100).to_i
+
+        if percent_complete > @progress[resource]
+
+          @progress[resource] = percent_complete
+
+          if percent_complete % interval == 0
```

Comment:
Yup, but I'm not sure it matters. I think this is fine as a best effort, just to give people some idea of what's happening for really big files.